lib/highlight.js

```diff
@@ -1,72 +1,73 @@
 'use strict';
 
-var hljs = require('highlight.js/lib/highlight');
-var Entities = require('html-entities').XmlEntities;
-var entities = new Entities();
-var alias = require('../highlight_alias.json');
+const hljs = require('highlight.js/lib/highlight');
+const Entities = require('html-entities').XmlEntities;
+const entities = new Entities();
+const alias = require('../highlight_alias.json');
 
-function highlightUtil(str, options) {
+function highlightUtil(str, options = {}) {
   if (typeof str !== 'string') throw new TypeError('str must be a string!');
-  options = options || {};
 
-  var useHljs = options.hasOwnProperty('hljs') ? options.hljs : false;
-  var gutter = options.hasOwnProperty('gutter') ? options.gutter : true;
-  var wrap = options.hasOwnProperty('wrap') ? options.wrap : true;
-  var firstLine = options.hasOwnProperty('firstLine') ? +options.firstLine : 1;
-  var caption = options.caption;
-  var mark = options.hasOwnProperty('mark') ? options.mark : [];
-  var tab = options.tab;
+  const {
+    gutter = true,
+    caption,
+    mark = [],
+    tab,
+    lang
+  } = options;
+
+  const useHljs = options.hasOwnProperty('hljs') ? options.hljs : false;
+
+  let { wrap = true, firstLine = 1 } = options;
+  firstLine = Number(firstLine);
 
   hljs.configure({ classPrefix: useHljs ? 'hljs-' : ''});
 
-  var data = highlight(str, options);
+  const { value, language } = highlight(str, options);
 
   if (useHljs && !gutter) wrap = false;
 
-  var before = useHljs ? '<pre><code class="hljs ' + options.lang + '">' : '<pre>';
-  var after = useHljs ? '</code></pre>' : '</pre>';
+  const before = useHljs ? `<pre><code class="hljs ${lang}">` : '<pre>';
+  const after = useHljs ? '</code></pre>' : '</pre>';
 
-  if (!wrap) return useHljs ? before + data.value + after : data.value;
+  if (!wrap) return useHljs ? before + value + after : value;
 
-  var lines = data.value.split('\n');
-  var numbers = '';
-  var content = '';
-  var result = '';
-  var line;
+  const lines = value.split('\n');
+  let numbers = '';
+  let content = '';
 
-  for (var i = 0, len = lines.length; i < len; i++) {
-    line = lines[i];
+  lines.forEach((line, i) => {
     if (tab) line = replaceTabs(line, tab);
-    numbers += '<span class="line">' + (firstLine + i) + '</span><br>';
+    numbers += `<span class="line">${firstLine + i}</span><br>`;
     content += formatLine(line, firstLine + i, mark, options);
-  }
+  });
 
-  result += '<figure class="highlight' + (data.language ? ' ' + data.language : '') + '">';
+  let result = `<figure class="highlight${language ? ` ${language}` : ''}">`;
```

Comment:
In the future maybe we can go into some of these ternary expressions and make them more explicit. That was there before so I didn't want to block it for this PR.

Reply:
@tcrowe thank review!
As you can see, I set it explicitly.